Updating cognitiveservices-luis-authoring SDK

ahmedabuamra opened the pull request:
All lines of code in this PR are auto-generated except updating the README.md file.

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/README.md:
```diff
@@ -15,38 +15,38 @@ npm install @azure/cognitiveservices-luis-authoring
+import { LUISAuthoringClient } from "@azure/cognitiveservices-luis-authoring";
+
+
+let subscriptionId = "b15ebe3a1ec446a08f8021fe6f95f0f6";
```
In your other PR for authoring, you use `let subscriptionId = "<luis-authoring-key>";`
Can we be consistent and use `let subscriptionId = "<luis-runtime-key>";` or whatever makes sense instead of the actual id?

ahmedabuamra replied:
Both of runtime and authoring sdks use luis authoring key to authenticate using `@azure/ms-rest-azure-js` package. so It makes more sense to use `let subscriptionId = "<luis-authoring-key>";` in both READMEs.

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/lUISAuthoringClient.ts:
```diff
@@ -34,8 +30,8 @@ class LUISAuthoringClient extends LUISAuthoringClientContext {
-  constructor(endpoint: string, credentials: msRest.ServiceClientCredentials, options?: msRest.ServiceClientOptions) {
-    super(endpoint, credentials, options);
+  constructor(credentials: msRest.ServiceClientCredentials, endpoint: string, options?: msRest.ServiceClientOptions) {
+    super(credentials, endpoint, options);
```
Was this change made manually or did the code generator do this?

ahmedabuamra replied:
It's the code generator.

ramya-rao-a replied:
Oh perfect, then one of the issues I had with luis-authoring regarding the order of the arguments being wrong is fixed by the code generator!

ahmedabuamra commented:
What I have changed:

- Added the copyrights manually.
- Replaced `subscriptionId` with `authoringKey` as @ChrisHMSFT suggested.
- Changed the `Impression` in README manually.
- Replaced the comment on versionId with @ramya-rao-a 's.

ChrisHMSFT commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/README.md:
```diff
@@ -15,38 +15,45 @@ npm install @azure/cognitiveservices-luis-authoring
+
+// change "westus" in the endpoint url based on your region, check the following link
+// https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-regions
+const client = new LUISAuthoringClient(creds, "https://westus.api.cognitive.microsoft.com/");
```
I would suggest we do not hardcode the endpoint in the README. We have seen a lot of first-use developers just copy and paste the snippets without checking the comments and this can lead to a confusing failure.

ChrisHMSFT commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/README.md:
```diff
@@ -15,38 +15,45 @@ npm install @azure/cognitiveservices-luis-authoring
+import { LUISRuntimeClient } from "@azure/cognitiveservices-luis-runtime";
+
+let authoringKey = "<luis-authoring-key>";
+const creds = new CognitiveServicesCredentials(subscriptionId);
```
Should the input here be authoringKey instead of subscriptionId?

Also - our standard in the quick start documentation is for the keys to all be stored as environment variables

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/README.md:
```diff
@@ -83,7 +92,7 @@ See https://github.com/Azure/ms-rest-browserauth to learn how to authenticate to
         const skip = 1;
         const take = 1;
-        client.features.listPhraseLists(appId, versionId, skip, take).then((result) => {
+        client.features.listApplicationVersionPatternFeatures(appId, versionId, skip, take).then((result) => {
```
```suggestion
        client.features.listApplicationVersionPatternFeatures(appId, versionId, { skip, take).then((result) => {
```

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/package.json:
```diff
@@ -25,13 +25,13 @@
     "uglify-js": "^3.4.9"
   },
-  "homepage": "https://github.com/azure/azure-sdk-for-js/tree/master/sdk/cognitiveservices/cognitiveservices-luis-authoring",
+  "homepage": "https://github.com/Azure/azure-sdk-for-js",
```
```suggestion
  "homepage": "https://github.com/azure/azure-sdk-for-js/tree/master/sdk/cognitiveservices/cognitiveservices-luis-authoring",
```

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/rollup.config.js:
```diff
@@ -21,9 +21,6 @@ const config = {
     banner: `/*
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/operations/apps.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/operations/azureAccounts.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/operations/examples.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/operations/features.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/operations/index.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/operations/model.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/operations/pattern.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/operations/permissions.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/operations/settings.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/operations/train.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/src/operations/versions.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copy right text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/package.json:
```diff
@@ -4,8 +4,8 @@
   "description": "LUISAuthoringClient Library with typescript type definitions for node.js and browser.",
   "version": "2.1.0",
```
```suggestion
  "version": "4.0.0",
```

ahmedabuamra commented:
@kayousef , @ramya-rao-a 
I've updated the version and added `"autoPublish": true` .

ramya-rao-a commented:
Since we have decided to get an update published with this PR approved, can we also get the browser sample updated?

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/README.md:
```diff
@@ -15,82 +15,89 @@ npm install @azure/cognitiveservices-luis-authoring
+      let authoringKey = process.env["luis-authoring-key"];
+      const creds = new CognitiveServicesCredentials(authoringKey);
+      const region = "<your-region>";
+      const client = new LUISAuthoringClient(
```
```suggestion
      const client = new Azure.CognitiveservicesLuisAuthoring.LUISAuthoringClient(
```

ramya-rao-a replied:
@ahmedabuamra This change is still pending.

ahmedabuamra replied:
sorry I missed it.

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/README.md:
```diff
@@ -15,82 +15,89 @@ npm install @azure/cognitiveservices-luis-authoring
     <title>@azure/cognitiveservices-luis-authoring sample</title>
-    <script src="node_modules/@azure/ms-rest-js/dist/msRest.browser.js"></script>
-    <script src="node_modules/@azure/ms-rest-browserauth/dist/msAuth.js"></script>
+    <script src="node_modules/@azure/ms-rest-azure-js/dist/cognitiveServicesCredentials.js"></script>
```
```suggestion
```

The `@azure/ms-rest-azure-js` package does not have a browser bundle and I have logged https://github.com/Azure/ms-rest-azure-js/issues/126 to track this issue.

Until that issue is fixed, we can use the `ApiKeyCredentials` instead

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-authoring/README.md:
```diff
@@ -15,82 +15,89 @@ npm install @azure/cognitiveservices-luis-authoring
-        const take = 1;
-        client.features.listPhraseLists(appId, versionId, skip, take).then((result) => {
+      let authoringKey = process.env["luis-authoring-key"];
+      const creds = new CognitiveServicesCredentials(authoringKey);
```
Since the `@azure/ms-rest-azure-js` package does not have a browser bundle yet, we should use the `ApiKeyCredentials` from `@azure/ms-rest-js` instead.

```suggestion
      const creds = new ApiKeyCredentials({ inHeader: { 'Ocp-Apim-Subscription-Key': authoringKey } });
```
